runelite-client/src/main/java/net/runelite/client/plugins/party/PartyPlugin.java

```diff
@@ -512,4 +553,42 @@ public void keyReleased(KeyEvent keyEvent)
 			hotkeyDown = false;
 		}
 	}
+
+	private void fetchAndSyncPlayerName()
+	{
+		clientThread.invoke(() ->
```

Comment:
I dont think you need to do this on clientThread

Reply:
I invoke this on client thread so that it keeps running until name returns something (or one of the other states happens). Sometimes the name will return null even when logged in.